components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/store/connector/RDMBSConnector.java

```diff
@@ -351,6 +360,34 @@ public void updateAssignmentAndState(Map<String, String> tasks) throws TaskCoord
         }
     }
 
+    /**
+     * Update the state of task.
+     *
+     * @param taskName     Name of the task.
+     * @param updatedState Updated state.
+     * @param destinedId   Destined Node Id.
+     * @return True if update is successful.
+     * @throws TaskCoordinationException when something goes wrong while updating.
+     */
+    public boolean updateTaskState(String taskName, CoordinatedTask.States updatedState, String destinedId)
```

Comment:
Why are we passing both the name and the destinedId? In case if the assignment has changed in the meantime?

Reply:
yes , to handle some edge cases. If everything is idea, assignment shouldn't have changed. If it is changed, we shouldn't be running it locally.